### PR TITLE
Fix spawner behaviour on failing controller activation or deactivation

### PR DESCRIPTION
**Pull request by saikishor**
Before the fix proposed in this PR, the spawner reports the controller activation as successful, if the activation is failed at the last stage after calling activate_controllers. This PR addresses the issue by proposing a fix and this should now properly fail at the spawner end as well

Before the fix:

```
1: [INFO] [1734338012.820800005] [test_controller_manager]: Activating controllers: [ ctrl_with_joint1_and_joint2_command_interfaces ]
1: [ERROR] [1734338012.829138606] [test_controller_manager]: Resource conflict for controller 'ctrl_with_joint1_and_joint2_command_interfaces'. Command interface 'joint1/position' is already claimed.
1: [INFO] [1734338012.830469222] [spawner_ctrl_with_joint1_and_joint2_command_interfaces]: Configured and activated ctrl_with_joint1_and_joint2_command_interfaces
```

With the proposed fix:
```
1: [INFO] [1734338233.806072093] [test_controller_manager]: Activating controllers: [ ctrl_with_joint1_and_joint2_command_interfaces ]
1: [ERROR] [1734338233.814172942] [test_controller_manager]: Resource conflict for controller 'ctrl_with_joint1_and_joint2_command_interfaces'. Command interface 'joint1/position' is already claimed.
1: [ERROR] [1734338233.814291503] [test_controller_manager]: Could not activate controller : 'ctrl_with_joint1_and_joint2_command_interfaces'
1: [ERROR] [1734338233.817168509] [spawner_ctrl_with_joint1_and_joint2_command_interfaces]: Failed to activate controller : ctrl_with_joint1_and_joint2_command_interfaces
```

### CODE REVIEW

File: controller_manager/test/test_spawner_unspawner.cpp [hunk 1]
```diff
@@ -437,6 +437,106 @@ TEST_F(TestLoadController, spawner_test_with_wildcard_entries_with_no_ctrl_name)
...
+
+  EXPECT_EQ(
+    call_spawner(
+      "ctrl_with_joint1_and_joint2_command_interfaces -c test_controller_manager "
+      "--controller-manager-timeout 1.0 "
+      "-p " +
+      test_file_path),
+    256)
+    << "Should fail as the ctrl_with_joint1_command_interface and "
+       "ctrl_with_joint2_command_interface are  active";
```
Comment (christophfroehlich) [inline controller_manager/test/test_spawner_unspawner.cpp:492]:
```suggestion
       "ctrl_with_joint2_command_interface are active";
```

**Review by christophfroehlich — APPROVED** [review]
LGTM (At least the test you've written makes totally sense.)

**Review by christophfroehlich — APPROVED** [review]

**Review by bmagyar — DISMISSED** [review]
Thank you!!

**Review by christophfroehlich — APPROVED** [review]
LGTM

**Pull request merged**